content/en/apps/guides/hosting/4.x/upgrading.md

```diff
@@ -0,0 +1,238 @@
+
+### 0. Ensure all clients have successfully synced
+This is not necessary. Should this be a step?
+
+### 1. Install CHT data migration tool
+
+Open your terminal and run these commands. They will create a new directory, download a docker compose file and download the required docker image. 
+```shell
+mkdir -p ~/couchdb-migration/ 
+cd ~/couchdb-migration/ 
+curl -s -o ./docker-compose.yml https://github.com/medic/couchdb-migration/blob/main/docker-compose.yml
+docker-compose up
+```
+
+For the following steps, the tool needs access to your CouchDb installation. To provide this access, you will need to provide a URL that includes authentication. 
+Additionally, if your CouchDb runs in docker, the tool needs to be added to the same docker network in order to access protected endpoints:
+
+```shell
+export COUCH_URL=http://admin:pass@127.0.0.1:5984
+```
+or 
+```shell
+export CHT_NETWORK=<docker-network-name>
+export COUCH_URL=http://admin:pass@docker-service-name:5984
+```
+
+For simplicity, you could store these required values in an `.env` file: 
+```shell
+cat > ${HOME}/couchdb-migration/.env << EOF
+CHT_NETWORK=cht-net
+COUCH_URL=http://admin:pass@127.0.0.1:5984
+EOF
+```
+
+### 2. Prepare CHT-Core 3.x installation for upgrading
+To minimize downtime when upgrading, it's advised to prepare the 3.x installation for the 4.x upgrade, and pre-index all views that ar required by 4.x.
+
+The migration tool provides a command which will download all 4.x views on your 3.x CouchDb installation, and initiate view indexing.
+
+```shell
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration pre-index-views 4.1.0
+```
+
+Once view indexing is finished, proceed with the next step.
+
+{{% alert title="Note" %}} If this step is omitted, 4.x API will fail to respond to requests until all views are indexed. Depending on the size of the database, this could take many hours, or even days. {{% /alert %}} 
+
+### 3. Save existent CouchDb configuration
+
+Some CouchDb configuration values must be ported from existent CouchDb to the 4.x installation. Store them in a safe location before shutting down 3.x CouchDb.
+
+##### a. CouchDB secret 
+Used in encrypting all CouchDb passwords and session tokens.
+##### b. CouchDb server uuid
+Used in generating replication checkpointer documents, which track where replication progress between every client and the server, and ensure that clients don't re-download or re-upload documents.
+
+Use the migration tool to obtain these values.
+```shell
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration get-env
+```
+
+### 4. Locate and backup CouchDb Data folder
+a) If running in MedicOS, [CouchDb data folder]({{< relref "apps/guides/hosting/3.x/self-hosting#backup" >}}) can be found at `/srv/storage/medic-core/couchdb/data`.
+
+b) If running a custom installation of CouchDb, data would be typically stored at `/opt/couchdb/data`.
+
+c) TODO when using orchestration / AWS 
+
+### 5. Launch 4.x CouchDb installation
+
```

Comment:
Add some instructions here to help people choose, but also, explain that they should pick either single node or multi node... at the moment it looks like you should do both!

Reply:
I added something, but with a TODO to link to a proper guide. I brought this up in our last infra call, but I don't think we ended up creating an issue. Do you think this information should live in this page or elsewhere?>